src/directives/obiba-file-download.js

```diff
@@ -22,17 +24,60 @@
         replace: true,
         scope: {
           url: '<',
+          formParams: '<',
           method: '@',
           encoding: '<'
         },
         link: function (scope, element) {
 
+          function moveUrlParamsToFormParams() {
+            var url = scope.url;
+
+            if (url) {
+              var index = url.indexOf('?');
+
+              if (index > -1) {
+                url.substring(index+1).split('&').forEach(function(param) {
+                  var parts = param.split('=');
+                  if (parts && !param.hasOwnProperty(parts[0])) {
+                    scope.formParams[parts[0]] = parts[1];
+                  }
+                });
+
+                scope.url = url.substring(0, index);
+              }
+            }
+          }
+
           function onClick(/*event*/) {
+            scope.formParams = scope.formParams || {};
+            moveUrlParamsToFormParams();
```

Comment:
Only when method is POST (because this directive still support GET with query params).

Reply:
Yes, will change.

Reply:
Done